src/compas/geometry/vector.py

```diff
@@ -148,11 +148,21 @@ def __add__(self, other):
     def __sub__(self, other):
         return Vector(self.x - other[0], self.y - other[1], self.z - other[2])
 
-    def __mul__(self, n):
-        return Vector(self.x * n, self.y * n, self.z * n)
-
-    def __truediv__(self, n):
-        return Vector(self.x / n, self.y / n, self.z / n)
+    def __mul__(self, other):
+        if isinstance(other, (int, float)):
+            return Vector(self.x * other, self.y * other, self.z * other)
+        elif isinstance(other, Vector):
+            return Vector(self.x * other[0], self.y * other[1], self.z * other[2])
```

Comment:
if `other` is `Vector`, why not `other.x` etc?
also, this is the same as `return self.dot(other)`

Reply:
you are very right.... another mistake of mine not careful about what copilot suggested! But this is not same with dot product tho, this returns element-wise multiplication, the result is still of a vector, not the sum of them